### PR TITLE
Add functionality to find out about locations from file numbers

### PR DESCRIPTION
**Pull request by oxinabox**
We need this to be able to set break points on File+LineNumbers

### CODE REVIEW

**Comment by oxinabox** [comment]
Required for #17 
I think this is good to go (passes tests locally).

@timholy would you care to take a glance over this?

I think it is suggestiver of functions that would be good to have in CodeTracking.jl.
It is using stuff from Revise and Rebuggerr

**Comment by oxinabox** [comment]
Contributing towards #6 


breadcrumbs

File: src/locate.jl [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+function containing_method(file, linenum)
+    for pkg_id in keys(Revise.pkgdatas)
+        meth = containing_method(pkg_id, file, linenum)
+        meth !== nothing && return meth
+    end
+end
+
+function sigt2meth(::Type{SIGT}) where SIGT
+    # TODO Revise.sig2t2methsig, should work, but sometimes seems not to
+    # so using this
```
Comment (timholy) [inline src/locate.jl]: Revise doesn't track all methods, unfortunately. `@eval`ed ones are a particular sore point.

**Comment by oxinabox** [comment]
For this to come into its own it really needs a  lot of other functionality, e.g. breaking midway  into a function.
But for now  it isn't useless in the current  master, so will look at merging soon

File: Project.toml [hunk 2]
```diff
@@ -5,13 +5,11 @@ version = "0.1.0"
...
 MacroTools = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
 Mocking = "78c3b35d-d492-501b-9361-3d52fe80e533"
 OrderedCollections = "bac558e1-5e72-5ebc-8fee-abe8a469f55d"
 Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
 
 [extras]
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
-
-[targets]
-test = ["Test"]
```
Comment (nickrobinson251) [inline Project.toml]: accidentally removed when doing a `rm`?

Comment (oxinabox) [reply Project.toml]: revolved in #23

**Comment by oxinabox** [comment]
Merged into #23.
Technically, there might be some value in merge this branch into master roughly as is.
But unless I get a request for it (which seems unlikely as I don't think anyone is using this yet?)
I am not going  to bother.

**Pull request closed without merging**